tests/unit/helpers/FixMovedQuestionConditionsTest.php

```diff
@@ -0,0 +1,219 @@
+<?php
+
+namespace ls\tests;
+
+class FixMovedQuestionConditions extends BaseTest
+{
+    public static function setUpBeforeClass(): void
+    {
+        parent::setupBeforeClass();
+
+        \Yii::import('application.helpers.common_helper', true);
+
+        // Import default survey
+        $filename = self::$surveysFolder . '/survey_fixMovedQuestionConditionsDefault.lss';
+        self::importSurvey($filename);
+
+        \Yii::app()->setConfig('sid', self::$surveyId);
+    }
+
+    /**
+     * Test fixing just one condition.
+     * This test uses the survey previously imported in setupBeforeClass.
+     */
+    public function testFixOneCondition()
+    {
+        $groups = self::$testSurvey->groups;
+        $questions = self::$testSurvey->allQuestions;
+
+        $conditions = \Condition::model()
+                            ->getSomeConditions(
+                                array('cfieldname'),
+                                'cqid = ' . $questions[0]->qid,
+                                null,
+                                null
+                            );
+
+        $firstQuestionConditions = $conditions->readAll();
+
+        $this->assertSame(
+            $firstQuestionConditions[0]['cfieldname'],
+            self::$surveyId . 'X' . $groups[0]->gid . 'X' . $questions[0]->qid,
+            'The cfieldname field is not correct.'
```

Comment:
Detail in the issue : _The cfieldname field is not correct : bad group before moving_ ?